gef-ui/server/utils/helpers.js

```diff
@@ -79,21 +83,125 @@ const validationErrorHandler = (errs, href = '') => {
+   if true, adds to array and returns array
+*/
+const facilitiesChangedToIssuedAsArray = (application) => {
+  const hasChanged = [];
+  application.facilities.items.map((facility) => {
+    if (facility.details.changedToIssued === true) {
+      const changed = {
+        name: facility.details.name,
+        id: facility.details._id,
+      };
+      hasChanged.push(changed);
+    }
+    return hasChanged;
+  });
+  return hasChanged;
+};
+
 /* Clean-Deep removes any properties with Null value from an Object. Therefore if all
   properties are Null, this leaves us with an Empty Object. isEmpty checks to see if the
   Object is empty or not. */
 const isEmpty = (value) => lodashIsEmpty(cleanDeep(value));
 
-const mapSummaryList = (data, itemsToShow, preview = false) => {
-  if (!data || lodashIsEmpty(data)) { return []; }
+/* summaryItemsConditions runs through the the table rows and decides if change/add should be added
+   on end of row.  On application details, all relevant rows are set to change/add if required.  On
+   application preview (once submitted to UKEF) - for facilities, certain rows are set to change add
+   and given personalised href.
+*/
+const summaryItemsConditions = (summaryItemsObj) => {
+  const {
+    preview, item, details, app, user, data,
+  } = summaryItemsObj;
+  const { id, href, shouldCoverStartOnSubmission } = item;
+  const value = typeof details[item.id] === 'number' || typeof details[item.id] === 'boolean' ? details[item.id].toString() : details[item.id];
+  const isCoverStartOnSubmission = id === 'coverStartDate' && shouldCoverStartOnSubmission;
+  // column keys to display change if facility has been changed to issued
+  const changedToIssueShow = id === 'name' || id === 'coverStartDate' || id === 'coverEndDate' || id === 'issueDate';
+  // column key to display add if facility not yet issued
+  const unissuedShow = id === 'hasBeenIssued';
+  // personalised href for facility to change to issued (once submitted to UKEF)
+  const unissuedHref = `/gef/application-details/${app._id}/unissued-facilities-change/${data.details._id}/about-facility`;
+  // array of facilities which have been changed to issued
+  const facilitiesChanged = facilitiesChangedToIssuedAsArray(app);
+
+  let summaryItems = [];
+  if (!preview) {
+    summaryItems = [
+      ...(href
+        ? [
+          {
+            href,
+            /* Clean-Deep removes any properties with Null value = require( an Object. Therefore if al)l
+           properties are Null, this leaves us with an Empty Object. isEmpty checks to see if the
+           Object is empty or not. */
+            text: `${isCoverStartOnSubmission || !isEmpty(value) ? 'Change' : 'Add'}`,
+            visuallyHiddenText: item.label,
+          },
+        ]
+        : []),
+    ];
+  } else if (app.status === CONSTANTS.DEAL_STATUS.UKEF_ACKNOWLEDGED && user.roles.includes('maker') && data.details.changedToIssued === true) {
```

Comment:
Condition to a constant/helper for readability?

Reply:
done